llm/main.py

```diff
@@ -0,0 +1,51 @@
+from flask import Flask, jsonify, abort, request
+from llm_functions import get_summary_api_function, get_tags_api_function
+import json
+from firebase_admin import initialize_app
+from firebase_functions import https_fn
+
+initialize_app()
+app = Flask(__name__)
+
+
+def is_intersection(keys, required_keys):
+    return (keys & required_keys) == required_keys
+
+
+@app.route("/summary", methods=["POST"])
+def summary():
+    body = json.loads(request.data)
+    # We require bill_id, bill_title, bill_text to exist as keys in the POST
+    if not is_intersection(body.keys(), {"bill_id", "bill_title", "bill_text"}):
+        abort(404, description="requires bill_id, bill_title, and bill_text")
```

Comment:
Note: currently this API takes a json payload like

```json
{
  "bill_id": "...",
  "bill_title": "...",
  "bill_text": "..." 
}
```

In the future, since this has access to firebase, you could also offer an API which takes the `bill_id` and returns the summary after looking up the title/text.